controlplane/kubeadm/internal/webhooks/kubeadmcontrolplanetemplate.go

```diff
@@ -65,46 +82,52 @@ func (r *KubeadmControlPlaneTemplate) ValidateCreate() (admission.Warnings, erro
 		)
 	}
 
-	spec := r.Spec.Template.Spec
+	spec := k.Spec.Template.Spec
 	allErrs := validateKubeadmControlPlaneTemplateResourceSpec(spec, field.NewPath("spec", "template", "spec"))
 	allErrs = append(allErrs, validateClusterConfiguration(spec.KubeadmConfigSpec.ClusterConfiguration, nil, field.NewPath("spec", "template", "spec", "kubeadmConfigSpec", "clusterConfiguration"))...)
```

Comment:
```suggestion
	allErrs = append(allErrs, validateClusterConfiguration(nil, spec.KubeadmConfigSpec.ClusterConfiguration,  field.NewPath("spec", "template", "spec", "kubeadmConfigSpec", "clusterConfiguration"))...)
```
Has to be adjusted as well as the order was flipped